blob/service.go

```diff
@@ -168,42 +167,47 @@ func (s *Service) GetAll(ctx context.Context, height uint64, namespaces []share.
 	}
 
 	var (
-		resultBlobs = make([][]*Blob, len(namespaces))
-		resultErr   = make([]error, len(namespaces))
-	)
+		blobs = make([]*Blob, 0)
 
-	for _, namespace := range namespaces {
-		log.Debugw("performing GetAll request", "namespace", namespace.String(), "height", height)
-	}
+		blobsCh = make(chan []*Blob, 1)
+		errCh   = make(chan error, 1)
+	)
 
-	wg := sync.WaitGroup{}
 	for i, namespace := range namespaces {
-		wg.Add(1)
 		go func(i int, namespace share.Namespace) {
-			defer wg.Done()
+			log.Debugw("retrieving all blobs from", "namespace", namespace.String(), "height", height)
 			blobs, err := s.getBlobs(ctx, namespace, header)
 			if err != nil {
-				resultErr[i] = fmt.Errorf("getting blobs for namespace(%s): %w", namespace.String(), err)
+				err = fmt.Errorf("getting blobs for namespace(%s): %w", namespace.String(), err)
+				errCh <- err
+				log.Debug(err)
 				return
 			}
 
 			log.Debugw("receiving blobs", "height", height, "total", len(blobs))
-			resultBlobs[i] = blobs
+			blobsCh <- blobs
 		}(i, namespace)
```

Comment:
Blobs were ordered before and now they are not. Could it be possible that some users depend on order of returned blobs? It would break API for such clients.

Reply:
Yes, but we've defined this PR as breaking. I'm also not sure that ordering can break smth.

Reply:
The order must be preserved; imagine asking for data in one order and getting it back in a completely different one. I would complain a lot about such an API personally

Reply:
reworked